flow/project.py

```diff
@@ -1577,55 +1528,37 @@ def _internal_call(self, func, name, *, cmd, with_job, directives):
                 func._flow_groups[self._parent_class] = {name}
                 return func
 
-            def with_directives(self, directives, name=None):
-                """Decorate a function to make it an operation with additional execution directives.
-
-                Directives can be used to provide information about required
-                resources such as the number of processors required for
-                execution of parallelized operations. For more information, see
-                :ref:`signac-docs:cluster_submission_directives`. To apply
-                directives to an operation that is part of a group, use
-                :meth:`.FlowGroupEntry.with_directives`.
-
-                Parameters
-                ----------
-                directives : dict
-                    Directives to use for resource requests and execution.
-                name : str
-                    The operation name. Uses the name of the function if None
-                    (Default value = None).
-
-                Returns
-                -------
-                function
-                    A decorator which registers the function with the provided
-                    name and directives as an operation of the
-                    :class:`~.FlowProject` subclass.
-                """
-                warnings.warn(
-                    "@FlowProject.operation.with_directives has been deprecated as of 0.22.0 and "
-                    "will be removed in 0.23.0. Use @FlowProject.operation(directives={...}) "
-                    "instead.",
-                    FutureWarning,
-                )
+            def _with_job(self, func):
+                base_aggregator = aggregator.groupsof(1)
+                if getattr(func, "_flow_aggregate", base_aggregator) != base_aggregator:
+                    raise FlowProjectDefinitionError(
+                        "The with_job keyword argument cannot be used with aggregation."
+                    )
 
-                def add_operation_with_directives(function):
-                    function._flow_directives = directives
-                    return self(function, name)
+                @functools.wraps(func)
+                def decorated(*jobs):
+                    with jobs[0] as job:
+                        if getattr(func, "_flow_cmd", False):
+                            return (
+                                f'trap "cd $(pwd)" EXIT && cd {job.ws} && {func(job)}'
+                            )
+                        else:
+                            return func(job)
 
-                return add_operation_with_directives
+                setattr(decorated, "_flow_with_job", True)
+                setattr(decorated, "_flow_cmd", getattr(func, "_flow_cmd", False))
```

Comment:
```suggestion
                if not hasattr(decorated, "_flow_cmd"):
                    decorated._flow_cmd = False
```

Reply:
decorated cannot have the attribute `_flow_cmd` by definition.
```suggestion
                decorated_flow_cmd = getattr(func, "_flow_cmd", False)
```

Reply:
My bad. I didn't catch that one argument was `func` and the other was `decorated`.

Reply:
Your suggestion looks correct here.